src/DataViews/FunctionsDataView.cpp

```diff
@@ -244,78 +242,55 @@ void FunctionsDataView::OnContextMenu(const std::string& action, int menu_index,
 }
 
 void FunctionsDataView::DoFilter() {
-  m_FilterTokens = absl::StrSplit(absl::AsciiStrToLower(filter_), ' ');
+  filter_tokens_ = absl::StrSplit(absl::AsciiStrToLower(filter_), ' ');
 
-#ifdef WIN32
-  ParallelFilter();
-#else
-  // TODO: port parallel filtering
-  std::vector<uint64_t> indices;
-  const std::vector<const FunctionInfo*>& functions(functions_);
-  for (size_t i = 0; i < functions.size(); ++i) {
-    const FunctionInfo* function = functions[i];
-    std::string name =
-        absl::AsciiStrToLower(orbit_client_data::function_utils::GetDisplayName(*function)) +
-        orbit_client_data::function_utils::GetLoadedModuleName(*function);
-
-    bool match = true;
-
-    for (std::string& filter_token : m_FilterTokens) {
-      if (name.find(filter_token) == std::string::npos) {
-        match = false;
-        break;
-      }
-    }
+  const size_t number_of_threads_available = thread_pool_->GetPoolSize();
 
-    if (match) {
-      indices.push_back(i);
-    }
-  }
+  constexpr size_t kMinimumNumberOfFunctionsPerThread = 64;
+  const size_t number_of_functions_per_thread =
+      std::max(kMinimumNumberOfFunctionsPerThread, functions_.size() / number_of_threads_available);
```

Comment:
ok, here is some additional thoughts, this is my personal preference since I might be overthinking it a bit.

We want number of tasks be somewhat greater then number of thread. The reason for it is that string have different length and search might be not evenly distributed between chunks. If we introduce `number_of_tasks = kSomeSmallButNotVerySmallPrime_Say7*number_of_threads` and use it here we give thread-pool more opportunities to balance the load better.

Reply:
Yes, might be. For me it's hard to judge whether that has more influence than the additional overhead of more tasks.

What kind of probability distribution can we assume for the length of function names? :sweat_smile: You're probably right with not evenly distributed since they are sorted alphanumerically, aren't they?

Reply:
Done.